server/infinispan/src/main/java/org/hawkular/apm/server/infinispan/InfinispanDynamicPropertyCache.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.apm.server.infinispan;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Logger;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.Resource;
+
+import org.hawkular.apm.api.services.ServiceLifecycle;
+import org.hawkular.apm.api.utils.PropertyUtil;
+import org.hawkular.apm.server.api.services.CacheException;
+import org.hawkular.apm.server.api.services.DynamicPropertyService;
+import org.hawkular.apm.server.api.services.ObservableProperty;
+import org.infinispan.Cache;
+import org.infinispan.manager.CacheContainer;
+
+/**
+ * @author Pavol Loffay
+ */
+public class InfinispanDynamicPropertyCache implements DynamicPropertyService, ServiceLifecycle {
+
+    private static final Logger log = Logger.getLogger(InfinispanSpanCache.class.getName());
+
+
+    protected static final String CACHE_NAME = "property";
+
+    @Resource(lookup = "java:jboss/infinispan/APM")
+    private CacheContainer cacheContainer;
+
+    private Cache<String, ObservableProperty> propertyCache;
+
+
+    @Override
+    @PostConstruct
+    public void init() {
+        // If cache container not already provisions, then must be running outside of a JEE
+        // environment, so create a default cache container
+        if (cacheContainer == null) {
```

Comment:
I might be wrong, but do all caches have a similar code? I remember seeing another Cache that had, to my memory, a very similar method. If so, might be worth having a `@Producer` somewhere, that accepts a `cacheName`, no? No need to add it to this PR, a JIRA would suffice, in case my memory is correct.